### PR TITLE
feat: add light mode

### LINKED ISSUE #271: [FEATURE] add light mode
### Description

Enable users to switch to dark/light modes easily.
This should be easy to implement with tailwindcss

### Screenshots

_No response_

### Additional information

_No response_

### 👀 Have you checked if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read and understood the rules in the [Contributing Guidelines](https://github.com/Dun-sin/Whisper/blob/main/CONTRIBUTING.md)

### Are you willing to work on this issue ?

None

### PR DESCRIPTION
**Pull request by ahmed-bentajhamyani**
# Fixes Issue

**My PR closes #271**

# 👨‍💻 Changes proposed(What did you do ?)

- Add light mode.

# ✔️ Check List (Check all the applicable boxes)

- [x] My code follows the code style of this project.
- [X] This PR does not contain plagiarized content.
- [X] The title and description of the PR is clear and explains the approach.

##  Note to reviewers

# 📷 Screenshots

Screenshot 2023-10-07 120520

### CODE REVIEW

**Comment by Dun-sin** [comment]
Default shouldn't be light mode

**Comment by Dun-sin** [comment]
Heading should say mode and not dark mode
Select should say the current mode and not dark mode either

**Comment by ahmed-bentajhamyani** [comment]
Noted, should I close this PR and open another one with those modifications right? "I'm a beginner in this open source thing".

**Comment by ahmed-bentajhamyani** [comment]
Done.

Screenshot 2023-10-07 164144

**Comment by Dun-sin** [comment]
> Default shouldn't be light mode

@ahmed-bentajhamyani

**Comment by Dun-sin** [comment]
It's not light mode when chatting
image

**Comment by ahmed-bentajhamyani** [comment]
Yes, bec this page doesn't work with me. It keeps telling me "Taking too long? No chat buddy is currently available :("

Screenshot 2023-10-08 104255

**Comment by ahmed-bentajhamyani** [comment]
Check now please !!

**Comment by Dun-sin** [comment]
> Yes, bec this page doesn't work with me. It keeps telling me "Taking too long? No chat buddy is currently available :("
> 
> Screenshot 2023-10-08 104255
> 

Did you read the Readme and follow how to install the project

**Comment by ahmed-bentajhamyani** [comment]
I got this issue in the vercel link too, so the problem is not related to the installation of the project. Though i found a way to modify the chat page please check my last commit.

**Comment by Dun-sin** [comment]
> I got this issue in the vercel link too, so the problem is not related to the installation of the project. Though i found a way to modify the chat page please check my last commit.

The original is different because it just means there's no one to connect you to, on your local device it's a different case. Read the Readme

**Comment by ahmed-bentajhamyani** [comment]
I did read it and i did all the steps.

Screenshot 2023-10-08 190922

And I pushed another commit that have light mode in the chatting page could you review it please?

**Comment by Dun-sin** [comment]
> I did read it and i did all the steps.
> Screenshot 2023-10-08 190922
> 
> And I pushed another commit that have light mode in the chatting page could you review it please?

you shouldn't have a problem matching with yourself then

**Comment by Dun-sin** [comment]
@ahmed-bentajhamyani you clearly didn't test your code, this is the last chance to fix this, after this time I will be closing and giving it to someone else. please take this seriously and if you don't want to let me know.

**Comment by ahmed-bentajhamyani** [comment]
I'm taking this seriously but I'm still having a problem to start a chat

Screenshot 2023-10-08 194122

Also this is my first time contributing to the open source, I'm still getting used to it. So please understand that this is just the mistakes of the newbies.

**Comment by ahmed-bentajhamyani** [comment]
Also I'm having a problem to Login with an account

Screenshot 2023-10-08 195119

When I click Login I got this error!!

**Comment by Dun-sin** [comment]
> Also I'm having a problem to Login with an account
> 
> Screenshot 2023-10-08 195119
> 
> When I click Login I got this error!!

This is why I said read the Readme🤦, there's an instruction for this, if you actually did you would know that, I don't appreciate being lied to

**Comment by ahmed-bentajhamyani** [comment]
I'm sorry it's my mistakes. I didn't understand the instructions very well, but I didn't lie.
Here should I change the textarea color or not ?

image

**Comment by Dun-sin** [comment]
we need a better color for white, that looks very ugly

**Comment by Dun-sin** [comment]
@ahmed-bentajhamyani please switch the colour for light mode to this -> `#edf7ff`

**Comment by ahmed-bentajhamyani** [comment]
image

image

What do you think 1 or 2?

**Comment by ahmed-bentajhamyani** [comment]
I resolved the conflict in my local repository, in the next push it will be resolved here

**Comment by Dun-sin** [comment]
> image
> 
> image
> 
> What do you think 1 or 2?
> 

I'm talking about the background

**Comment by ahmed-bentajhamyani** [comment]
> I'm sorry it's my mistakes. I didn't understand the instructions very well, but I didn't lie.
> Here should I change the textarea color or not ?
> 
> image
> 

Aaah ok. Just bec I asked about the textarea.

**Comment by ahmed-bentajhamyani** [comment]
image

Like that ?

**Comment by ahmed-bentajhamyani** [comment]
@Dun-sin ??

**Comment by ahmed-bentajhamyani** [comment]
@Dun-sin I didn't get a reply from you?

**Comment by Dun-sin** [comment]
> image
> 
> Like that ?

Yes

**Comment by ahmed-bentajhamyani** [comment]
Ok so know should I change the white background in all the pages right ?

**Comment by Dun-sin** [comment]
> Ok so know should I change the white background in all the pages right ?

Put the light mode variable in tailwind and then update it for all light mode

**Comment by ahmed-bentajhamyani** [comment]
@Dun-sin please check the last commits I pushed in this PR.

**Comment by ahmed-bentajhamyani** [comment]
Hey @Dun-sin, please check the last commits I pushed in this PR, it's ready for review.

**Comment by Dun-sin** [comment]
@ahmed-bentajhamyani stop commenting, I will check it when I can.

Don't do that in the future either, maintainers will get back to you when they can

**Comment by ahmed-bentajhamyani** [comment]
Ok, i 'm sorry

File: client/tailwind.config.js [hunk 1]
```diff
@@ -10,6 +11,7 @@ module.exports = {
                 secondary: '#162938',
                 highlight: '#FF9F1C',
                 red: '#FF3A46',
+                'alice-blue': '#EDF7FF'
```
Comment (Dun-sin) [inline client/tailwind.config.js:14]:
```suggestion
                'light': '#EDF7FF'
```

Comment (Dun-sin) [inline client/tailwind.config.js:14]:
```suggestion
                light: '#f4faff'
```

File: client/src/pages/Start.jsx [hunk 2]
```diff
@@ -26,9 +26,11 @@ const Start = () => {
         <div
             className={createClassesFromArray([
                 centerElement,
-                'bg-primary',
+                'bg-alice-blue',
```
Comment (Dun-sin) [inline client/src/pages/Start.jsx:29]:
```suggestion
                'bg-light',
```

File: client/src/styles/dialog.css [hunk 3]
```diff
@@ -8,7 +8,7 @@
     }
 
     .dialog-container {
-        @apply z-50 rounded-lg fixed top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 m-0 bg-secondary w-[90%] md:w-auto md:max-w-md px-6 py-6;
+        @apply z-50 rounded-lg fixed top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 m-0 bg-alice-blue dark:bg-secondary w-[90%] md:w-auto md:max-w-md px-6 py-6;
```
Comment (Dun-sin) [inline client/src/styles/dialog.css:11]:
```suggestion
        @apply z-50 rounded-lg fixed top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 m-0 bg-light dark:bg-secondary w-[90%] md:w-auto md:max-w-md px-6 py-6;
```

File: client/src/pages/Settings.jsx [hunk 4]
```diff
@@ -56,12 +61,77 @@ const Searching = () => {
         updateTmpSettings(newSettings);
     };
 
+    const { darkMode, setDarkMode } = useDarkMode();
+
     return (
         <div
-            className="flex justify-center items-center flex-col min-w-[calc(100%-120px)] p-2 gap-5 bg-primary min-h-full"
+            className="flex justify-center items-center flex-col min-w-[calc(100%-120px)] p-2 gap-5 bg-alice-blue dark:bg-primary min-h-full"
```
Comment (Dun-sin) [inline client/src/pages/Settings.jsx:68]:
```suggestion
            className="flex justify-center items-center flex-col min-w-[calc(100%-120px)] p-2 gap-5 bg-light dark:bg-primary min-h-full"
```

File: client/src/pages/Profile.jsx [hunk 5]
```diff
@@ -83,20 +83,22 @@ const Profile = () => {
     return (
         <div
             className={createClassesFromArray(
-                'bg-primary',
+                'bg-alice-blue',
```
Comment (Dun-sin) [inline client/src/pages/Profile.jsx:86]:
```suggestion
                'bg-light',
```

File: client/src/pages/Login.jsx [hunk 6]
```diff
@@ -24,14 +24,14 @@ const Login = () => {
 
     return (
         <div
-            className={`bg-primary h-[100vh] w-[100vw] text-white ${centerStuffs}`}
+            className={`bg-alice-blue dark:bg-primary h-[100vh] w-[100vw] text-primary dark:text-white ${centerStuffs}`}
```
Comment (Dun-sin) [inline client/src/pages/Login.jsx:27]:
```suggestion
            className={`bg-light dark:bg-primary h-[100vh] w-[100vw] text-primary dark:text-white ${centerStuffs}`}
```

File: client/src/pages/ComingSoon.jsx [hunk 7]
```diff
@@ -5,14 +5,16 @@ const ComingSoon = () => {
     return (
         <div
             className={createClassesFromArray(
-                'bg-primary',
+                'bg-alice-blue',
```
Comment (Dun-sin) [inline client/src/pages/ComingSoon.jsx:8]:
```suggestion
                'bg-light',
```

File: client/src/components/BuddyMatcher.jsx [hunk 8]
```diff
@@ -217,9 +217,9 @@ const BuddyMatcher = () => {
     }, [app.currentChatId]);
 
     return app.isSearching || !app.currentChatId ? (
-        <div className="flex w-full justify-center items-center min-h-[calc(100vh-70px)] flex-col bg-primary">
+        <div className="flex w-full justify-center items-center min-h-[calc(100vh-70px)] flex-col bg-alice-blue dark:bg-primary">
```
Comment (Dun-sin) [inline client/src/components/BuddyMatcher.jsx:220]:
```suggestion
        <div className="flex w-full justify-center items-center min-h-[calc(100vh-70px)] flex-col bg-light dark:bg-primary">
```

File: client/src/components/Anonymous.jsx [hunk 9]
```diff
@@ -165,12 +165,14 @@ const Anonymous = ({ onChatClosed }) => {
         <div
             className={createClassesFromArray([
                 centerItems,
-                'bg-primary',
+                'bg-alice-blue',
```
Comment (Dun-sin) [inline client/src/components/Anonymous.jsx:168]:
```suggestion
                'bg-light',
```

File: client/src/components/NavBar.jsx [hunk 10]
```diff
@@ -80,7 +80,6 @@ const NavBar = ({ className }) => {
                 'md:flex-col',
                 'flex-row',
                 'justify-center',
```
Comment (Dun-sin) [inline client/src/components/NavBar.jsx:82]:
```suggestion
                'justify-center',
                'shadow-[0_0_100px_0_rgba(0,0,0,1)]',
```

**Review by Dun-sin — APPROVED** [review]
You did great, thanks for contributing, I hope you will stick around and continue to contribute to this project.

Consider giving this project a star and joining the community discord server if you haven't for more resources and opportunities to connect with others. [👉🏽here👈🏽](https://discord.com/invite/ufcysW9q23)

**Pull request merged**